hdf5-types/src/dyn_value.rs

```diff
@@ -115,19 +115,76 @@ impl From<DynInteger> for DynValue<'_> {
 }
 
 #[derive(Copy, Clone, PartialEq)]
-pub enum DynScalar {
-    Integer(DynInteger),
+pub enum DynFloat {
```

Comment:
Bit of an edge case but we could add `#[cfg_attr(not(feature = "f16"), non_exhaustive)]
`